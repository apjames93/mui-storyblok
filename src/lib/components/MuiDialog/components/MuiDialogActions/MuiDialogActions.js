import React from 'react';
import PropTypes from 'prop-types';
import DialogActions from '@material-ui/core/DialogActions';
import { validComponents } from '../../../../utils/customProps';
import Storyblok from '../../../../utils/Storyblok';
import MuiIconButtonRedirect from '../../../MuiIconButtonRedirect/MuiIconButtonRedirect';
import MuiIconButtonHref from '../../../MuiIconButtonHref/MuiIconButtonHref';
import MuiIconButtonDownload from '../../../MuiIconButtonDownload/MuiIconButtonDownload';
import MuiButtonRedirect from '../../../MuiButtonRedirect/MuiButtonRedirect';
import MuiButtonHref from '../../../MuiButtonHref/MuiButtonHref';
import MuiButtonDownload from '../../../MuiButtonDownload/MuiButtonDownload';
import { renderComponents } from '../../../../utils/customComponents';

const MuiDialogActions = ({
  content,
  rootClass,
}) => {
  const components = {
    MuiIconButtonRedirect,
    MuiIconButtonHref,
    MuiIconButtonDownload,
    MuiButtonRedirect,
    MuiButtonHref,
    MuiButtonDownload,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <DialogActions
      className={styles.root}
    >
      {content.map((component, key) => renderComponents(components, component, key))}
    </DialogActions>
  );
};

export default MuiDialogActions;

MuiDialogActions.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /**
    'MuiIconButtonRedirect',
    'MuiIconButtonHref',
    'MuiIconButtonDownload',
    'MuiButtonRedirect',
    'MuiButtonHref',
    'MuiButtonDownload',
    */
  content(props, propName, componentName) {
    const components = [
      'MuiIconButtonRedirect',
      'MuiIconButtonHref',
      'MuiIconButtonDownload',
      'MuiButtonRedirect',
      'MuiButtonHref',
      'MuiButtonDownload',
    ];
    return validComponents(props, propName, componentName, components);
  },
};

MuiDialogActions.defaultProps = {
  rootClass: [],
  content: [],
};
