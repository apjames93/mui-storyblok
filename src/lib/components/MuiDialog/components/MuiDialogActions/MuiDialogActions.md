##### MuiDialogActions has the ability to render custom components.