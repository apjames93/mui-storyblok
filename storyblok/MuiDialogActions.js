module.exports = {
  "name": "MuiDialogActions",
  "display_name": null,
  "updated_at": "2020-07-14T17:36:26.550Z",
  "schema": {
    "rootClass": {
      "type": "options"
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiIconButtonRedirect",
        "MuiIconButtonHref",
        "MuiIconButtonDownload",
        "MuiButtonRedirect",
        "MuiButtonDownload",
        "MuiButtonHref",
        "GoogleConversions"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiDialogActions",
  "component_group_uuid": null
}