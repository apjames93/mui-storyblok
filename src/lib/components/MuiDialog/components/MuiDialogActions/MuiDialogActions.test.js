import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiDialogActions from './MuiDialogActions';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "MuiDialogActions", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    content: [{
      _editable: '<!--#storyblok#{"name": "MuiButtonRedirect", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiButtonRedirect',
      redirectRoute: 'page-welcome',
      button: [{
        _editable: '<!--#storyblok#{"name": "MuiButton", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiButton',
        buttonText: 'buttonText',
      }],
    }],
  };
  const comp = shallow(<MuiDialogActions {...props} />);

  return { comp, props };
}

describe('<MuiDialogActions />', () => {
  it('renders MuiDialogActions', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiDialogActions {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
