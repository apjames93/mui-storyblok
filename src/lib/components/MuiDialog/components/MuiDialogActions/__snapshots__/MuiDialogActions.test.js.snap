// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiDialogActions /> snapshot 1`] = `
<div
  className="MuiDialogActions-root makeStyles-root-3 MuiDialogActions-spacing"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-root-4 MuiButton-textSecondary MuiButton-fullWidth"
    data-testid="muiButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      buttonText
    </span>
  </button>
</div>
`;
